engine/src/widget.cpp

```diff
@@ -812,12 +812,9 @@ IO_stat MCWidget::save(IO_handle p_stream, uint4 p_part, bool p_force_ext, uint3
     if (m_widget != nil)
         MCWidgetOnSave(m_widget, &t_rep);
     else
-        t_rep = m_rep;
-    
-    // If the rep is nil, then an error must have been thrown, so we still
-    // save, but without any state for this widget.
-    if (*t_rep == nil)
-        t_rep = MCValueRetain(kMCNull);
+		// If the rep is nil, then an error must have been thrown, so we still
+		// save, but without any state for this widget.
+		&t_rep = (m_rep != nil ? m_rep : MCValueRetain(kMCNull));
```

Comment:
Could you use 
```
else if (m_rep != nil)
    t_rep = m_rep
else
{
    // If the rep is nil, then an error must have been thrown, so we still
    // save, but without any state for this widget.
    t_rep = kMCNull
}
```
- It's in the style guide to avoid the ternary operator where possible
- I'm uneasy about single line else that comes after a comment - even though it's fine it makes it easy to make a mistake in the future and add a line after it
- For an `MCAutoValueRef` and friends, assignment using `=` retains the target value (indeed the original was over-retaining `kMCNull`)

Reply:
Thanks @livecodeali